### ISSUE TITLE
`max-statements-per-line`: complains on `if` when `curly`-rule adds missing `{`

### ISSUE THREAD

**Issue by jgburet** [eslint-stylistic/eslint-stylistic#295]
### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

Hey,
I'm not sure if this should be added to https://github.com/eslint-stylistic/eslint-stylistic/issues/211 or if that's a different issue

The rule `max-statements-per-line` (with the default `max: 1`) complains on `if (v) { console.log(); }`.
These braces are added when the rule `curly` is present with `"all"` or `"multi", "consistent"` (when the `else` has multiple statements within its braces).

I'm expecting the rule to accept the block because it contains only a single statement.

Reproduction:
```yml
extends:
  - eslint:recommended
  - plugin:@typescript-eslint/strict-type-checked
  - plugin:@typescript-eslint/stylistic-type-checked
parser: "@typescript-eslint/parser"
parserOptions:
  ecmaVersion: latest
  sourceType: module
  project:
    - "tsconfig.json"
    - "./**/tsconfig.json"
plugins:
  - "@typescript-eslint"
  - "@stylistic"
rules:
  curly:
    - warn
    - "multi"
    - "consistent"
  "@stylistic/max-statements-per-line":
    - error
    - max: 1
```

```typescript
    if (v) { console.log("hyel"); }
    else {
        console.log("");
        console.log( );
    }
```

### Reproduction

https://stackblitz.com/edit/github-pwcljb?file=eslint.config.js

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests

**Comment by m-shum** [comment]
It would be really nice if we could have this rule auto-fixed the same way it's autofixed on stackblitz.
